### PR TITLE
Update http Response code attribute names

### PR DESCRIPTION
**Pull request by meiao**
### Overview
This adds the `http.statusCode` and `http.statusText` to spans and transactions.

### Related Github Issue
#225 

### Checks

[y] Are your contributions backwards compatible with relevant frameworks and APIs?
[n] Does your code contain any breaking changes? Please describe. 
[n] Does your code introduce any new dependencies? Please describe.

### CODE REVIEW

File: instrumentation-test/src/main/java/com/newrelic/agent/introspec/internal/ExternalRequestImpl.java [hunk 1]
```diff
@@ -23,15 +25,19 @@
 
     private String library;
     private String operation;
+    private Integer statusCode;
```
Comment (twcrone) [inline instrumentation-test/src/main/java/com/newrelic/agent/introspec/internal/ExternalRequestImpl.java:28]: So do these need to be mutable after construction?  If not, can we make them final?

File: instrumentation/aws-java-sdk-s3-2.0/src/test/java/com/agent/instrumentation/awsjavasdks3/AmazonS3AsyncApiTest.java [hunk 2]
```diff
@@ -103,35 +103,35 @@ public void testListBuckets() {
...
 
     @Test
     public void testGetBucketLocation() {
         createBucketNoTxn();
         getBucketLocation();
-        assertMetrics("getBucketLocation");
+        assertMetrics("getBucketLocation", 200);
     }
 
 // This test passes but it consistently takes 60s to complete
```
Comment (twcrone) [inline instrumentation/aws-java-sdk-s3-2.0/src/test/java/com/agent/instrumentation/awsjavasdks3/AmazonS3AsyncApiTest.java:137]: So do we ever uncomment and run this test or should it go?

File: instrumentation/http-async-client-4/src/main/java/org/apache/http/nio/protocol/HttpAsyncResponseConsumer_Instrumentation.java [hunk 3]
```diff
@@ -84,4 +88,19 @@ public void failed(Exception ex) {
         }
         Weaver.callOriginal();
     }
+
+    private Integer getStatusCode() {
+        if (httpResponse != null && httpResponse.getStatusLine() != null)
+        {
```
Comment (twcrone) [inline instrumentation/http-async-client-4/src/main/java/org/apache/http/nio/protocol/HttpAsyncResponseConsumer_Instrumentation.java:94]: This just looks weird to me.

File: instrumentation/okhttp-4.4.0/src/test/java/com/nr/agent/instrumentation/okhttp44/OkHttp44Test.java [hunk 4]
```diff
@@ -151,6 +153,8 @@ public void testCat() throws Exception {
         ExternalRequest externalRequest = externalRequests.iterator().next();
         assertEquals(1, externalRequest.getCount());
         assertEquals(host, externalRequest.getHostname());
+        assertEquals(Integer.valueOf(200), externalRequest.getStatusCode());
+        assertEquals("OK ", externalRequest.getStatusText()); // the test server does return the trailing space, this client does not trim it
```
Comment (twcrone) [inline instrumentation/okhttp-4.4.0/src/test/java/com/nr/agent/instrumentation/okhttp44/OkHttp44Test.java:157]: These are good examples of comments on the "Why".

File: newrelic-api/src/main/java/com/newrelic/api/agent/HttpParameters.java [hunk 5]
```diff
@@ -91,6 +109,26 @@ public String getProcedure() {
         return procedure;
     }
 
+    /**
+     * Returns the HTTP status code for the call.
+     *
+     * @return the status code for the call, null if not available
+     * @since 6.5.0
```
Comment (twcrone) [inline newrelic-api/src/main/java/com/newrelic/api/agent/HttpParameters.java:116]: Is this still going to be `6.5.0`?

**Review by twcrone — APPROVED** [review]

File: instrumentation-test/src/main/java/com/newrelic/agent/introspec/ExternalRequest.java [hunk 6]
```diff
@@ -17,6 +17,10 @@
 
     String getLibrary();
 
+    Integer getStatusCode();
+
+    String getStatusText();
```
[the remark this answers is not in the document]
Comment (meiao) [reply instrumentation-test/src/main/java/com/newrelic/agent/introspec/ExternalRequest.java:22]: Yes, it is in the agent spec.

File: instrumentation/aws-java-sdk-s3-2.0/src/test/java/com/agent/instrumentation/awsjavasdks3/AmazonS3AsyncApiTest.java (hunk 2 above)
Comment (meiao) [reply instrumentation/aws-java-sdk-s3-2.0/src/test/java/com/agent/instrumentation/awsjavasdks3/AmazonS3AsyncApiTest.java:137]: We can uncomment it now that we have GE.

File: newrelic-api/src/main/java/com/newrelic/api/agent/HttpParameters.java (hunk 5 above)
Comment (meiao) [reply newrelic-api/src/main/java/com/newrelic/api/agent/HttpParameters.java:116]: nope. This is how long ago I started working on this.

**Pull request merged**